docker/src/main/java/clocker/docker/entity/util/DockerUtils.java

```diff
@@ -304,7 +304,7 @@ private static String getContainerNameFromPlan(Entity target) {
     public static String getTargetAddress(Entity source, Entity target) {
         boolean local = source.sensors().get(SoftwareProcess.PROVISIONING_LOCATION).equals(target.sensors().get(SoftwareProcess.PROVISIONING_LOCATION));
         List networks = target.sensors().get(SdnAttributes.ATTACHED_NETWORKS);
-        if (local && (networks != null && networks.size() > 0)) {
+        if ( local ) {
```

Comment:
Without this change you end up having to explicitly set the mapped host/port when linking to another container, for example wordpress you'd have to set the following:
```

  env:
    WORDPRESS_DB_HOST: $brooklyn:component("db").attributeWhenReady("mapped.docker.port.3306")
```
rather than being able to rely on getting at the db via the `mysql` alias alone - ie the local address added to `/etc/hosts` and using the default unmapped db port.